src/hierarchy.ts

```diff
@@ -275,10 +275,12 @@ export async function group(store?: Store | string, path: string | null = null,
  * @param chunkStore Store or path to directory in file system or name of zip file.
  * @param cacheAttrs If `true` (default), user attributes will be cached for attribute read operations
  *   If False, user attributes are reloaded from the store prior to all attribute read operations.
- *
+ * @param ipfsClient IPFS client
+ * @param cid IPFS CID
  */
-export async function openGroup(store?: Store | string, path: string | null = null, mode: PersistenceMode = "a", chunkStore?: Store, cacheAttrs = true) {
-    store = normalizeStoreArgument(store);
+export async function openGroup(store?: Store | string, path: string | null = null, mode: PersistenceMode = "a", chunkStore?: Store, cacheAttrs = true, ipfsClient?: any,
+cid?: any,) {
```

Comment:
since we want to push this upstream and ensure high quality of code I think we should add types instead using any 

- https://discuss.ipfs.tech/t/how-to-use-the-typescript-types-with-ipfs-http-client/10564/5 (direct link to a .d.ts file https://gitlab.com/kelp_digital/ipfs-files-management/-/blob/main/src/types/ipfs-http-client.d/index.d.ts )
- https://github.com/ipfs/js-ipfs/issues/2945 
- https://github.com/ipfs-examples/js-ipfs-types-use-ipfs-from-ts/blob/main/src/main.ts 


`import type { CID } from 'multiformats/cid'`

Reply:
seems like they need to import `ipfs-http-client` for the one typescript declaration or am i reading that wrong?

Reply:
on a second google I found this https://github.com/bluelovers/ws-ipfs/tree/master/packages/ipfs-types which should have the ipfsclient type among others but yes that could be the alternative what you have above

Reply:
it still imports `ipfs-core-types` which has a size of 201kb. The package is already 5.73 mb. is that something we want to do?

Reply:
resolved by doing a barebones type definition